Assets/MixedRealityToolkit.SDK/Features/UX/Interactable/Scripts/Interactable.cs

```diff
@@ -510,8 +589,51 @@ private void RemovePointer(IMixedRealityPointer pointer)
             pointers.Remove(pointer);
         }
 
+        /// <summary>
+        /// Access the PointerManager with all the interactive pointer data.
+        /// </summary>
+        /// <returns></returns>
+        public InteractablePointerDataManager GetPointerManager()
+        {
+            return pointerDataManager;
+        }
+
         #endregion PointerManagement
 
+        #region SourceManagement
+
+        /// <summary>
+        /// Adds a source to sources
+        /// </summary>
+        /// <param name="source"></param>
+        private void AddSource(SourceStateEventData source)
+        {
+            if (!sources.Contains(source))
+            {
+                sources.Add(source);
+            }
+        }
+
+        /// <summary>
+        /// Removes a source, lost
+        /// </summary>
+        /// <param name="source"></param>
+        private void RemoveSource(SourceStateEventData source)
+        {
+            sources.Remove(source);
+        }
+
+        /// <summary>
```

Comment:
nit: Curious why this is being implemented as a method and not a property. In MRTK we tend to use properties.

Reply:
Just old habit, I'll update this. Also, I should the source stuff to it's own PR. It could be self contained and we can discuss if this is needed, because it doesn't work consistently using the source handlers unless the button is set to global. This is part of an older request to support HoloLens interaction states, Interactive, Observation, Observation Targeted and Targeted, where source status is required.

Reply:
These have been updated, or removed.